### ISSUE TITLE
Disable all interrupts when entering debugging session

### ISSUE THREAD

**Issue by scottc11** [scottc11/ok-degree#9]

**Comment by cameronfyfe** [comment]
What's the motivation for this?  Normally it should be mostly fine.

What's the debugging configuration/link firmware you're using currently and what's behavior you're seeing that you don't want to have?

I used to have the best luck using jlink firmware both with cortex-debug in vscode and with ST's IDE. st-link firmware is okay but seems to have weird delays, and openocd was usually my last choice, though might've been just because I had weird experiences with it, not knowing fully how to set it up optimally.  If you're using the openocd config that's currently in vscode, I would maybe first try flashing j-link and see if that makes things any better

**Comment by scottc11** [comment]
So one of my interrupts increments an array index value which I use to record data into an array as well as read data out of the array for processing. This is problematic when stepping through the code with debugger because that index value will keep getting incremented / rolled over.

SO, in an ideal world, if its possible ( I think it is 🤔) when the debugger hits a breakpoint in the code, it disables all interrupts from firing, so I can properly track how this ever important index value gets handled. You know?

It would be nice to try a different debugging probe like jlink. I really don't even know the difference though. Would it use the same mini JTAG connector as the ST-LINK? Would it use OpenOCD?

My debugging configuration is slightly unkown to me. I believe I am using OpenOCD and a "GDB-Server" 🤷‍♂️

**Comment by cameronfyfe** [comment]
Yeah, I think this will be resolved when you switch over to the J-Link firmware, once you have it flashed let me know if you're having trouble running it and we can get it setup, it's pretty much the same as setting up openocd in the configuration in launch.json.  There should be an example in the docs for that extension

**Comment by cameronfyfe** [comment]
The behavior that I'm used to with J-Link sessions is interrupts fire when you have it running, but do not fire when you have the CPU halted on a breakpoint or are just stepping through lines/instructions
